src/stratis_cli/_actions/_top.py

```diff
@@ -153,9 +172,32 @@ def add_data_devices(namespace):
             .search(managed_objects)
         )
 
+        cache = [
```

Comment:
I think it makes sense to use ```frozenset``` to make it obvious that these are a bunch of set operations.

```
blockdevs = frozenset(namespace.blockdevs)
cache = frozenset(...)
already_cache =  blockdevs intersection cache

unchanged = blockdevs intersection data
# etc.
```

Also, I think ```unchanged``` would be clearer as ```already_data``` to go with ```already_cache```.

Reply:
I've pushed a bug fix and the name change so far. Still working on the rest.

Reply:
This is now resolved. I've switched to `frozenset`, convert to lists of strings for output, and use the appropriate set operations.